### PR TITLE
feat: support gradle

### PR DESCRIPTION
**Pull request by miiila**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Adds `gradle support` for a callgraph creation

### Notes for the reviewer

~We need to compare callgraph outputs before saying final 🆗~

Testing callgraphs compared are exactly the same.

### More information

- [Jira ticket FLOW-416](https://snyksec.atlassian.net/browse/FLOW-416)

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -5,18 +5,22 @@ import { getCallGraph } from './java-wrapper';
 import { Graph } from 'graphlib';
 import { timeIt, getMetrics, Metrics } from './metrics';
 import { CallGraphGenerationError } from './errors';
+import { glob } from './promisified-fs-glob';
+import { MissingTargetFolderError } from './errors';
```
Comment (darmalovan) [inline lib/index.ts:9]: Can be amended to line 7

File: lib/index.ts [hunk 2]
```diff
@@ -5,18 +5,22 @@ import { getCallGraph } from './java-wrapper';
...
 export async function getCallGraphMvn(
   targetPath: string,
   timeout?: number,
 ): Promise<Graph> {
   try {
-    const classPath = await timeIt('getMvnClassPath', () =>
-      getClassPathFromMvn(targetPath),
-    );
+    const [classPath, targets] = await Promise.all([
+      timeIt('getGradleClassPath', () => getClassPathFromMvn(targetPath)),
```
Comment (darmalovan) [inline lib/index.ts:18]: Probably should `getMavenClassPath` ?

File: test/lib/java-wrapper.test.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ test('classes per jar mapping is created', async () => {
 
 test('not target folder throw error', async () => {
   expect(
-    getTargets('some-bogus-folder-that-does-not-exist'),
+    getTargets('some-bogus-folder-that-does-not-exist', 'mvn'),
```
Comment (darmalovan) [inline test/lib/java-wrapper.test.ts:29]: Do we have a gradle test?

Comment (miiila) [reply test/lib/java-wrapper.test.ts:29]: It's not needed for non-existing upper level folder.

Comment (darmalovan) [reply test/lib/java-wrapper.test.ts:29]: Meant more if we have things that test gradle?

Comment (miiila) [reply test/lib/java-wrapper.test.ts:29]: Not sure we understand each other - this is not maven specific test.

In general, coverage is roughly the same.

File: .circleci/config.yml [hunk 4]
```diff
@@ -208,13 +232,13 @@ workflows:
...
               ignore:
                 - master
-      - test-integration-windows:
-          name: Integration Tests Windows
-          context: nodejs-install
-          filters:
-            branches:
-              ignore:
-                - master
+#- test-integration-windows:
```
Comment (darmalovan) [inline .circleci/config.yml:235]: Maybe adding comment on why it's commented:? :)

Comment (miiila) [reply .circleci/config.yml:235]: It is in commit message, but sure :-)

File: test/lib/java-wrapper.test.ts [hunk 5]
```diff
@@ -1,9 +1,9 @@
 import * as path from 'path';
 import {
   getClassPerJarMapping,
-  getTargets,
   getCallGraphGenCommandArgs,
 } from '../../lib/java-wrapper';
+import { getTargets } from '../../lib/index';
```
Comment (darmalovan) [inline test/lib/java-wrapper.test.ts:6]: Small thing is that this file doesn't test `/index` - may be worth moving?

**Review by darmalovan — APPROVED** [review]

File: test/lib/java-wrapper.test.ts (hunk 5 above)
Comment (miiila) [reply test/lib/java-wrapper.test.ts]: Addressed.

**Pull request merged**